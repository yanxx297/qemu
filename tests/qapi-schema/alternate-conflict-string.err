tests/qapi-schema/alternate-conflict-string.json:4: Anonymous union 'MyUnion' member 'two' can't be distinguished from member 'one'
