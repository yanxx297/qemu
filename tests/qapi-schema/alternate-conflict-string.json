# FIXME: we should reject anonymous unions with multiple string-like branches
{ 'enum': 'Enum',
  'data': [ 'hello', 'world' ] }
{ 'union': 'MyUnion',
  'discriminator': {},
  'data': { 'one': 'str',
            'two': 'Enum' } }
