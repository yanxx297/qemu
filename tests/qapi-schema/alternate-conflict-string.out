[OrderedDict([('enum', 'Enum'), ('data', ['hello', 'world'])]),
 OrderedDict([('union', 'MyUnion'), ('discriminator', OrderedDict()), ('data', OrderedDict([('one', 'str'), ('two', 'Enum')]))])]
[{'enum_name': 'Enum', 'enum_values': ['hello', 'world']},
 {'enum_name': 'MyUnionKind', 'enum_values': None}]
[]
